test/request.js

```diff
@@ -275,45 +276,64 @@ describe('Request', () => {
 
     describe('active()', () => {
 
-        it('exits handler early when request is no longer active', async () => {
+        it('exits handler early when request is no longer active', { retry: true }, async (flags) => {
+
+            const onCleanup = [];
+            flags.onCleanup = async () => {
+
+                for (const cleanup of onCleanup) {
+                    await cleanup();
+                }
+            };
+
+            let testComplete = false;
+            onCleanup.unshift(() => {
+
+                testComplete = true;
+            });
 
             const server = Hapi.server();
-            const team = new Teamwork.Team();
-            const team2 = new Teamwork.Team();
+            const enterHandlerTeam = new Teamwork.Team();
+            const leaveHandlerTeam = new Teamwork.Team();
 
-            let rounds = 0;
```

Comment:
Ah, you essentially replaced the rounds limit with a while loop with a `testComplete` failsafe.